haunted-houses/src/test/java/cz/muni/fi/pa165/hauntedhouses/SpecterDaoTests.java

```diff
@@ -0,0 +1,319 @@
+        Assert.assertEquals(s1.getStartOfHaunting(),foundSpecters.get(0).getStartOfHaunting());
+        Assert.assertEquals(s1.getEndOfHaunting(),foundSpecters.get(0).getEndOfHaunting());
+
+        foundSpecters = specterDao.getSpectersByHouse(house2);
+        Assert.assertEquals(foundSpecters.size(),1);
+        Assert.assertEquals(s2.getName(), foundSpecters.get(0).getName());
+        Assert.assertEquals(s2.getDescription(), foundSpecters.get(0).getDescription());
+        Assert.assertEquals(s2.getStartOfHaunting(),foundSpecters.get(0).getStartOfHaunting());
+        Assert.assertEquals(s2.getEndOfHaunting(),foundSpecters.get(0).getEndOfHaunting());
+
+        foundSpecters = specterDao.getSpectersByHouse(house3);
+        Assert.assertEquals(foundSpecters.size(),0);
+    }
+
+    @Test
+    public void createAndGetByAbilitySpecterTest() {
+        Ability ability1 = new Ability();
+        ability1.setName("AbilityName1");
+        ability1.setDescription("AbilityDescription1");
+        abilityDao.createAbility(ability1);
+
+        Ability ability2 = new Ability();
+        ability2.setName("AbilityName2");
+        ability2.setDescription("AbilityDescription2");
+        abilityDao.createAbility(ability2);
+
+        Ability ability3 = new Ability();
+        ability3.setName("AbilityName3");
+        ability3.setDescription("AbilityDescription3");
+        abilityDao.createAbility(ability3);
+
+        s1.setAbilities(Arrays.asList(new Ability[]{ability1,ability2}));
+        s2.setAbilities(Arrays.asList(new Ability[]{ability2}));
+
+        specterDao.createSpecter(s1);
+        specterDao.createSpecter(s2);
+
+        List<Specter> foundSpecters = specterDao.getSpectersByAbility(ability1);
+        Assert.assertEquals(foundSpecters.size(),1);
+        Assert.assertEquals(s1.getName(), foundSpecters.get(0).getName());
+        Assert.assertEquals(s1.getDescription(), foundSpecters.get(0).getDescription());
+        Assert.assertEquals(s1.getStartOfHaunting(),foundSpecters.get(0).getStartOfHaunting());
+        Assert.assertEquals(s1.getEndOfHaunting(),foundSpecters.get(0).getEndOfHaunting());
+
+        foundSpecters = specterDao.getSpectersByAbility(ability2);
+        Assert.assertEquals(foundSpecters.size(),2);
+        Assert.assertEquals(s1.getName(), foundSpecters.get(0).getName());
+        Assert.assertEquals(s1.getDescription(), foundSpecters.get(0).getDescription());
+        Assert.assertEquals(s1.getStartOfHaunting(),foundSpecters.get(0).getStartOfHaunting());
+        Assert.assertEquals(s1.getEndOfHaunting(),foundSpecters.get(0).getEndOfHaunting());
+        Assert.assertEquals(s2.getName(), foundSpecters.get(1).getName());
+        Assert.assertEquals(s2.getDescription(), foundSpecters.get(1).getDescription());
+        Assert.assertEquals(s2.getStartOfHaunting(),foundSpecters.get(1).getStartOfHaunting());
+        Assert.assertEquals(s2.getEndOfHaunting(),foundSpecters.get(1).getEndOfHaunting());
```

Comment:
Je skutočne zaručené, že sa dostanú do toho zoznamu práve v tomto poradí? Asi nechceme testovať aj konkrétne poradie Specters po zavolaní getSpectersByAbility.

Reply:
Chtěl jsem si to trochu zjednodušit tím, že to pořadí znám, ale je pravda, že takové pořadí není požadované a správně mohou být entity v listu asi i prohozené. Opravím to.